spacelift/resource_named_webhook_secret_header.go

```diff
@@ -0,0 +1,152 @@
+		},
+
+		Schema: map[string]*schema.Schema{
+			"webhook_id": {
+				Type:        schema.TypeString,
+				Description: "ID of the stack on which the environment variable is defined",
+				Required:    true,
+				ForceNew:    true,
+			},
+			"key": {
+				Type:             schema.TypeString,
+				Description:      "key for the header",
+				Required:         true,
+				ValidateDiagFunc: validations.DisallowEmptyString,
+				ForceNew:         true,
+			},
+			"value": {
+				Type:             schema.TypeString,
+				Description:      "value for the header",
+				DiffSuppressFunc: suppressValueChange,
+				Required:         true,
+				Sensitive:        true,
+				ForceNew:         true,
+			},
+		},
+	}
+}
+
+func resourceNamedWebhookSecretHeaderCreate(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	var mutation struct {
+		WebhooksIntegration struct {
+			ID string `graphql:"id"`
+		} `graphql:"namedWebhooksIntegrationSetHeaders(id: $id, input: $input)"`
+	}
+
+	webhookID, _ := d.GetOk("webhook_id")
+	variables := map[string]interface{}{
+		"id": toID(webhookID),
+		"input": structs.NamedWebhookHeaderInput{
+			Entries: []structs.KeyValuePair{
+				{
+					Key:   d.Get("key").(string),
+					Value: d.Get("value").(string),
+				},
+			},
+		},
+	}
+
+	var ret diag.Diagnostics
+	if err := meta.(*internal.Client).Mutate(ctx, "NamedWebhookSecretHeaderSet", &mutation, variables); err != nil {
+		ret = diag.Errorf("could not set secret header: %v", internal.FromSpaceliftError(err))
+	}
+
+	d.SetId(fmt.Sprintf("%s/%s", d.Get("webhook_id"), d.Get("key")))
+
+	return append(ret, resourceNamedWebhookSecretHeaderRead(ctx, d, meta)...)
+}
+
+func resourceNamedWebhookSecretHeaderRead(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	idParts := strings.SplitN(d.Id(), "/", 2)
+	if len(idParts) != 2 {
+		return diag.Errorf("unexpected resource ID: %s", d.Id())
+	}
+
+	var query struct {
+		Webhook *structs.NamedWebhooksIntegration `graphql:"namedWebhooksIntegration(id: $id)"`
+	}
+
+	resourceID, variableKey := idParts[0], idParts[1]
+
+	variables := map[string]interface{}{"id": graphql.ID(resourceID)}
+	if err := meta.(*internal.Client).Query(ctx, "GetNamedWebhook", &query, variables); err != nil {
+		return diag.Errorf("could not query for named webhook: %v", err)
+	}
+
+	if query.Webhook == nil {
+		d.SetId("")
+		return nil
+	}
+
+	wh := query.Webhook
+
+	found := false
+	for _, sh := range wh.SecretHeaders {
+		if sh == variableKey {
+			d.Set("key", sh)
+			found = true
+			break
+		}
+	}
+
+	// If we didn't fail adding the key this should never happen.
+	if !found {
+		d.SetId("")
+	}
+
+	return nil
+}
+
+func resourceNamedWebhookSecretHeaderDelete(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	idParts := strings.SplitN(d.Id(), "/", 2)
+	if len(idParts) != 2 {
+		return diag.Errorf("unexpected resource ID: %s", d.Id())
+	}
+
+	var mutation struct {
```

Comment:
This will fail if the named webhook is gone, and this is not unusual because if you're deleting both things at the same time, the webhook will go 1st.

There's a few things we can do here:
- in a delete, perform a read 1st, and if the webhook is gone, call it a success - not ideal because it's not atomic/transactional;
- rescue an API error that will be reported when the webhook is gone, call it a success - not great, not terrible;
- change the mutation API to return a `nil` response when the webhook is gone, and change this code to take it into account;

Reply:
> change the mutation API to return a nil response when the webhook is gone, and change this code to take it into account;

I don't think there is anything to take in to account here? Basically if it doesn't return an error, the code just continues and sets the ID to be empty without parsing anything extra.